lib/rust/ensogl/core/src/debug/monitor.rs

```diff
@@ -804,67 +730,14 @@ impl Sampler for Fps {
         "Frames per second"
     }
     fn value(&self) -> f64 {
-        self.value
+        self.stats.fps()
     }
     fn check(&self) -> ValueCheck {
-        self.value_check
+        self.check_by_threshold(FPS_WARNING_THRESHOLD, FPS_ERROR_THRESHOLD)
```

Comment:
I will start with noting some necessary background before I reply to (1) and (2):

Please make sure to observe, that:

 - the calculations of FPS, that were previously done in the `Fps::begin` function (and stored in the `Fps::value` field), are now performed in the `StatsData::begin_frame` function in `frame_stats.rs` (and stored in the `StatsData::fps` field).
 - The same applies to storing the value of the `time` argument, representing the frame begin time, previously stored in the `Fps::begin_time` field, currently stored in the `StatsData::begin_time` field (intended to be renamed to `frame_begin_time`).
 - The `StatsData::begin_frame` function thus must be, and is, called at frame begin, which is the same moment in time as where the `Fps::begin` function was called previously.
 - Also, notably, in `StatsData::end_frame`, the `fps` value is not touched, same as it was not touched in the `end` function (nor is the `begin_time` value being touched there).
 
Now, with the above observations in mind:

 1. In the original implementation, the value check was performed in the `Fps::begin` function, but still *after* the `self.value` field calculation and assignment. As such, it stayed "valid" and cached starting from a call to the `Fps::begin` function, until the next call to the `Fps::begin` function. In the new implementation, the value check is no longer cached. However, a call to `Fps::check` done after `StatsData::begin_frame` does the value check based on `self.value()`, which returns `self.stats.fps()`, which returns the value of the `StatsData::fps` field calculated in `StatsData::begin_frame`. As such, it will return the same value between calls to `StatsData::begin_frame`, as previously between calls to `Fps::begin` - with an exception of point (2) below.
 2. Thank you for noticing and pointing out the "first value ever gathered" case - this I missed indeed. I thus plan to restore the `Sampler::begin` function and the `Fps::begin` function, and add an extra frame counter field in the `Fps` struct, and an extra `if` in the `Fps::check` function based on the frame counter, accounting for the case of the first value ever gathered, so as to return `ValueCheck::Correct` in this case (as was done in the previous implementation).